### ISSUE TITLE
Ignore some kind of elements in revapi-java 0.13.1

### ISSUE THREAD

**Issue by surli** [revapi/revapi#63]
Based on #60 I tried to add the following rule in our config:
```json
"ignore": [
        {
          "regex": true,
          "code": ".*",
          "elementKind": "^(?!interface)$",
          "justification": "Per policy, only interface types are checked for API changes."
        }
      ]
```

But I still obtain the following errors:
```
[ERROR] "code": "java.method.addedToInterface",
[ERROR] "new": "method spoon.reflect.visitor.chain.CtQuery spoon.reflect.factory.Factory::createQuery(java.lang.Object)",
[ERROR] "package": "spoon.reflect.factory",
[ERROR] "classSimpleName": "Factory",
[ERROR] "methodName": "createQuery",
[ERROR] "elementKind": "method",
[ERROR] "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR] "code": "java.field.serialVersionUIDUnchanged",
[ERROR] "old": "field spoon.reflect.factory.FactoryImpl.serialVersionUID",
[ERROR] "new": "field spoon.reflect.factory.FactoryImpl.serialVersionUID",
[ERROR] "package": "spoon.reflect.factory",
[ERROR] "classSimpleName": "FactoryImpl",
[ERROR] "fieldName": "serialVersionUID",
[ERROR] "elementKind": "field",
[ERROR] "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
```

So maybe the first one is justified because it's an `addedToInterface` cause (but then the rule is not clear), but the second one should completely be ignored.

Not sure it's a bug, maybe it's not yet fully supported :)

**Comment by metlos** [comment]
`"elementKind": "^(?!interface)$"` -> `"elementKind": "^(?!interface).*$"` but I think I was overly eager at getting this in and didn't think through what this feature enables and what it doesn't.

As you point out above, a "method added to interface" should I think be part of your reports but if you filter out every change that is not directly on an interface, it means that you loose `java.method.addedToInterface` et al, because those are changes on the method level, not the interfaces...

**Comment by metlos** [comment]
This should be doable if the ignore transform was able to also inspect the parents of the element. That is doable, but I need to think about how to formulate the restrictions like above in some meaningful manner.

**Comment by metlos** [comment]
Implemented as part of the fix for #17.
